packages/ozone-api/ozone-api-upload/src/ozone-api-upload.ts

```diff
@@ -350,16 +357,16 @@ export class UploadFileRequest implements XMLHttpRequestLike {
 		request.body = JSON.stringify(info)
 
 		return request.sendRequest()
-				.then((xhr: XMLHttpRequest) => {
-					const response: UploadEndResult = data as UploadEndResult
-					response.uploadFileId = xhr.response.file
-					return response
-				})
+			.then((xhr: XMLHttpRequest) => {
+				const response: UploadEndResult = data as UploadEndResult
+				response.uploadFileId = xhr.response.file
+				return response
+			})
 	}
-	async _waitForTask(uploadEndResult: UploadFileId): Promise < string | undefined > {
+
+	_waitForTask(taskId: string): Promise<TaskResult | undefined> {
 		const taskClient = getDefaultClient().taskClient()
-		const taskHandler = taskClient.waitForTask<TaskResult>(uploadEndResult.uploadFileId)
-		const result = await taskHandler.waitResult
-		return result && result.mediaId
+		const taskHandler = taskClient.waitForTask<TaskResult>(taskId, { skipWaitingOnSubTask: true })
```

Comment:
Shouldn't that be an option to be passed to the module ? This lib is not only made to be used by flowr-admin, we should have the choice to wait for full completion.

Reply:
You're right. I'm making it an option to pass to the uploadFile function.
Do you know where it's used ? In the FlowR monorepo, I've only seen it used in flowr-admin.
The problem is that in the old implementation, uploadFile was returning a string (the media id) but in mine, it has to return both the mediaId and the subsequent tasks that are skipped.
I don't want to make two return types unless I'm forced to do so.

Reply:
The ozone components can be used by others projects like Flowr-Connect ( phone application .. ) or before it was used by Adobe Plugin.